src/Helpers.php

```diff
@@ -83,7 +83,8 @@ public static function build_url($protocol, $host, $base_path, $url)
             //drive: followed by a relative path would be a drive specific default folder.
             //not known in php app code, treat as abs path
             //($url[1] !== ':' || ($url[2]!=='\\' && $url[2]!=='/'))
-            if ($url[0] !== '/' && (strtoupper(substr(PHP_OS, 0, 3)) !== 'WIN' || (mb_strlen($url) > 1 && $url[0] !== '\\' && $url[1] !== ':'))) {
+            //Also respect the base path for the file:// protocol
+            if (mb_strtolower($protocol) == 'file://' || $url[0] !== '/' && (strtoupper(substr(PHP_OS, 0, 3)) !== 'WIN' || (mb_strlen($url) > 1 && $url[0] !== '\\' && $url[1] !== ':'))) {
```

Comment:
I don't think this is how we want to tweak the logic to better handle file-protocol support. The current logic already support a file-protocol base protocol or URL. Current logic prepends the base path if the URL isn't absolute, which is the expected behavior. With this change when the base protocol is "file://" Dompdf would _always_ prepend the base path. We only want to prepend the base path if the url is relative.

Reply:
I need to re-check this ... one by one. It may be a setting confusion ..